packages/components/src/responsive-wrapper/index.js

```diff
@@ -7,22 +7,31 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { cloneElement, Children } from '@wordpress/element';
+import { useResizeObserver } from '@wordpress/compose';
 
 function ResponsiveWrapper( {
 	naturalWidth,
 	naturalHeight,
 	children,
 	isInline = false,
 } ) {
+	const [
+		containerResizeListener,
+		{ width: containerWidth },
+	] = useResizeObserver();
 	if ( Children.count( children ) !== 1 ) {
 		return null;
 	}
 	const imageStyle = {
-		paddingBottom: ( naturalHeight / naturalWidth ) * 100 + '%',
+		paddingBottom:
+			naturalWidth < containerWidth
```

Comment:
We may want to be conscious of the fact that `containerWidth` will be `null` the first time the component renders, and in comparisons like this, `null` is treated effectively the same as `0`. I'm not sure based on the logic here if that's what we're expecting.